### ISSUE TITLE
How to disable the encryption to improve performance?

### ISSUE THREAD

**Issue by NavpreetDevpuri** [jpillora/chisel#384]
As mentioned in this #347, It is slow because of encryption. I don't have any security concerns as the user only use HTTPS, and the user is well aware of security. So, is there any way to just disable the encryption in order to improve performance? what changes should I make in the code?

**Comment by jpillora** [comment]
I don’t think it’s the encryption that slows it down, I believe it’s the
SSH protocol overhead combined with a lack of performance tuning throughout
the code base

Feel free to benchmark though, you might be able to edit the code to use a
null cipher - this will likely require forking the ssh package too…

**Comment by NavpreetDevpuri** [comment]
@jpillora Thanks for the quick reply, I will try.

**Comment by SuperSandro2000** [comment]
> It is slow because of encryption

No, that is not right like this. It is mostly slow because of the encapsulation of multiple protocols which generates quite some overhead. TLS is not adding much overhead to that.

**Comment by genofire** [comment]
I does understand this Issue - to have an option to disable SSH (and leave TLS / HTTPS / WebsocketS).

Is this possible?

**Comment by jpillora** [comment]
No SSH provides connection multiplexing (virtual TCP in TCP)

**Comment by rucciva** [comment]
To anyone looking for alternative, might i suggest https://github.com/rapiz1/rathole. It support tunneling via websocket. From my brief benchmark using iperf3, it can attain almost 2x the bandwith of ssh tunnel.
